### ISSUE TITLE
Training too slow

### ISSUE THREAD

**Issue by caiyu97** [mutianxu/GDANet#1]
Hello, I'm very interested in your work, but when I try to reproduce it, I find that the training is too slow. Even if I maximize the batchsize, it takes three Epochs a day. My GPU is Nvidia gtx1080ti 11G. What's the reason?

**Comment by mutianxu** [comment]
Hi,

Thanks for your interest on our work.

I'm wondering which task you met the problem, classification or segmentation?

Regards,
Mutian

**Comment by caiyu97** [comment]
classification

**Comment by mutianxu** [comment]
Hi @caiyu97 ,

I think your GPU is OK. We use the same GPU as yours but still train very fast, about 30-60sec/ep.

_Three Epochs a day_ is not expected if everything goes well.

Also, note that increasing batchsize will not decrease the training time since it will increase the time/iteration, thus I do not recommend you to do this.

So far I can make sure that it is not the problem of our code.

I suggest you:
1. Check your PyTorch version and CUDA version. 
2. Check your **CPU**_util by and **RAM** usage by running `top` or `htop`, which may also affect the training speed. (If CPU_util  or RAM usage of other programs are very large, the machine will be stuck.)

Regards,
Mutian

**Comment by caiyu97** [comment]
Thank you for your help. I've found my problem.Thank you very much for your prompt reply

**Comment by alfredtorres** [comment]
Hello,  I also encounter the problem when I reproducing the classification taks. My GPU is 1080Ti.
I find that most time consumption is during the loss.backward(). In addition, during training, my GPU and CPU are running at around 100%. I use the same code on your git.
My pytroch version is 1.7.1  & CUDA 11.0.
I am wondering what is the problem.

Regards,
Ziyu.
